gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/ApiPrimaryOwnerRemovalUpgrader.java

```diff
@@ -0,0 +1,201 @@
+package io.gravitee.rest.api.service.impl.upgrade;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.StringUtils.*;
+
+import io.gravitee.repository.exceptions.TechnicalException;
+import io.gravitee.repository.management.api.ApiRepository;
+import io.gravitee.repository.management.api.EnvironmentRepository;
+import io.gravitee.repository.management.api.GroupRepository;
+import io.gravitee.repository.management.api.MembershipRepository;
+import io.gravitee.repository.management.api.OrganizationRepository;
+import io.gravitee.repository.management.api.RoleRepository;
+import io.gravitee.repository.management.api.UserRepository;
+import io.gravitee.repository.management.api.search.ApiCriteria;
+import io.gravitee.repository.management.model.*;
+import io.gravitee.rest.api.model.permissions.SystemRole;
+import io.gravitee.rest.api.service.Upgrader;
+import io.gravitee.rest.api.service.common.UuidString;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.core.Ordered;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author GraviteeSource Team
+ */
+@Component
+public class ApiPrimaryOwnerRemovalUpgrader implements Upgrader, Ordered {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ApiPrimaryOwnerRemovalUpgrader.class);
+
+    private final RoleRepository roleRepository;
+
+    private final ApiRepository apiRepository;
+
+    private final MembershipRepository membershipRepository;
+
+    private final OrganizationRepository organizationRepository;
+
+    private final EnvironmentRepository environmentRepository;
+
+    private final UserRepository userRepository;
+
+    private final GroupRepository groupRepository;
+
+    @Value("${apis.primaryOwner.default:}")
+    private String defaultPrimaryOwnerId;
+
+    public ApiPrimaryOwnerRemovalUpgrader(
+        @Lazy RoleRepository roleRepository,
+        @Lazy ApiRepository apiRepository,
+        @Lazy MembershipRepository membershipRepository,
+        @Lazy OrganizationRepository organizationRepository,
+        @Lazy EnvironmentRepository environmentRepository,
+        @Lazy UserRepository userRepository,
+        @Lazy GroupRepository groupRepository
+    ) {
+        this.roleRepository = roleRepository;
+        this.apiRepository = apiRepository;
+        this.membershipRepository = membershipRepository;
+        this.organizationRepository = organizationRepository;
+        this.environmentRepository = environmentRepository;
+        this.userRepository = userRepository;
+        this.groupRepository = groupRepository;
+    }
+
+    @Override
+    public boolean upgrade() {
+        try {
+            Set<Organization> organizations = organizationRepository.findAll();
```

Comment:
Could it be an `OrganizationUpgrader` instead ?
That would benefit of the common organization loop.

Reply:
as discussed together, there is not yet OrganizationUpgrader in 3.15